gaia-media-controls.js

```diff
@@ -0,0 +1,291 @@
+;(function(define){'use strict';define(function(require,exports,module){
+
+/**
+ * Dependencies
+ */
+var Component = require('gaia-component');
+var MediaControlsImpl = require('./lib/media-controls-impl');
+
+var MediaControls = Component.register('gaia-media-controls', {
+  /**
+   * 'createdCallback' is called when the element is first created.
+   */
+  created: function() {
+    console.log('creating gaia-media-controls web component...');
+  },
+
+  attachTo: function(player) {
+
+    if (this.mediaPlayerImpl) {
+      throw new Error('A media player is already attached to the media controls component');
+    }
+
+    if (!this.shadowRoot) {
+      this.setupShadowRoot();
+    }
```

Comment:
If you don't call setupShadowRoot() when the element is first created, I worry that the user might see the controls flash into place after the rest of the DOM is rendered and the JS loads and calls attachTo().  Could you call setupShadowRoot() in the created() function instead?

Or is that already happening automatically in gaia-component somewhere?